isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/spring.adoc

```diff
@@ -0,0 +1,290 @@
+Zusätzlich erbt die Klasse von `SpringBootServletInitializer`, damit die Anwendung im Tomcat deployt werden kann.
+Die Konfigurationsklassen der Schichten werden per `@Import`-Annotation eingebunden.
+
+[[listing-applikationsklasse]]
+.Applikationsklasse der Anwendung
+[source,java]
+----
+package de.beispiel.if2anwendung;
+
+@Configuration
+@EnableAutoConfiguration
+@Import({ CoreConfig.class, PersistenceConfig.class, ServiceConfig.class })
+public class IsyFactApplication extends SpringBootServletInitializer {
+
+    @Override
+    protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
+        return application.sources(IsyFactApplication.class);
+    }
+}
+----
+
+Für jede Schicht wird im Package der Schicht (`<org>.<domäne>.<anwendung>.<schicht>`) parallel zu den Interfaces der Komponentenschnittstellen eine mit `@Configuration` annotierte <<listing-konfigurationsklasse,Konfigurationsklasse>> erstellt.
+Diese wird zusätzlich mit `@ComponentScan` annotiert.
+Annotationen, die querschnittliche Aspekte konfigurieren (z.B. `@EnableTransactionManagement`), werden an die Konfigurationsklasse der inhaltlich passenden Schicht geschrieben.
+
+[[listing-konfigurationsklasse]]
+.Konfigurationsklasse des Anwendungskerns
+[source,java]
+----
+package de.beispiel.if2anwendung.core;
+
+@Configuration
+@EnableTransactionManagement
+@ComponentScan
+public class CoreConfig {
+
+    @Bean
+    public BeispielBean beispielBean() {
+        // ...
+    }
+}
+----
+
+Die Spring-Beans der Komponenten werden mit der Annotation `@Component` bzw. deren Spezialisierungen (`@Repository`, `@Service`, `@Controller` etc.) versehen.
+Externe Spring-Beans, die nicht annotiert werden können, werden in der Konfigurationsklasse der Schicht per `@Bean`-Methoden konfiguriert.
+
+[[webapprootkey-konfigurieren]]
+=== webAppRootKey konfigurieren
+
+Der Kontextparameter `webAppRootKey` muss in der `web.xml` auf den eindeutigen Namen der Webanwendung gesetzt werden.
+
+[[listing-webAppRootKey]]
+.Konfiguration des webAppRootKey
+[source,xml]
+----
+<context-param>
+  <param-name>webAppRootKey</param-name>
+  <param-value>allgemeine-anwendung</param-value>
+</context-param>
+----
+
+*Hintergrund*: Spring speichert standardmäßig den Pfad zum Wurzelverzeichnis der Webanwendung im Webserver in der Kontextvariable `webapp.root`.
+Wenn mehrere Anwendungen gleichzeitig in einem Tomcat betrieben werden (bspw.
+in einer Entwicklungsumgebung), wird dieser Parameter durch die verschiedenen Anwendungen überschrieben.
+Dies kann zu ungewünschten Seiteneffekten führen.
+Ist der Kontextparameter `webAppRootKey` wie im obigen Beispiel gesetzt, wird der Pfad statt im Parameter `webapp.root`
+im Parameter `allgemeine-anwendung` abgelegt.
+Da jede Webanwendung einen eindeutigen Namen besitzt, und damit einen eigenen Kontextparameter verwendet, wird das Überschreiben vermieden.
+
+[[direkte-verwendung-des-spring-frameworks]]
+== Direkte Verwendung von Spring
```

Comment:
Geht es in diesem Kapitel um Vorgaben?
Wenn ja, dann sollte diese Kapitel in "Vorgaben zur direkten Verwendung von Spring" umbenannt werden